File: python/guess.py
import random
import sys

min = 1
max = 1000

if len(sys.argv) > 1 :
    max = int(sys.argv[1])

number = random.randint(min, max)

print('I have selected a number between %d and %d' % (min, max))
print('Please try to guess my number.')

guess_count = 0

while True : 
    guess = input('Your guess: ')

    try :
        guess = int(guess)
    except :
        print("That doesn't look like a number. Try again.")
        continue

    guess_count += 1

    if guess == number :
        print('You guess by number in %d guesses!' % (guess_count))
        break
    elif guess > number : 
        print('Your guess was too high. Try again.')
    else :
        print('Your guess was too low. Try again.')

File: c/guess.c
#include <stdio.h>
#include <sys/random.h>

int randnum(int max) {
    int val;

    // useful only on linux systems...
    getrandom(&val, sizeof(int), GRND_NONBLOCK);

    if (val < 0) {
        return -1 * val % max + 1;
    } else {
        return val % max + 1;
    }
}

int main(void) {
    int max = 1000;
    int number = randnum(max);
    int guessCount = 0;
    int guess;

    printf("I have selected a number between %d and %d.\n", 1, max);
    printf("Please guess my number.\n");

    while (1) {
        printf("Your guess: ");

        if ( scanf("%d", &guess) != 1 ) {
            int c;
            while ((c = getchar()) != EOF && c != '\n')
            {
                continue;
            }
            printf("That doesn't look like a valid number. Try again.\n");
            continue;
        }

        guessCount ++;

        if ( guess == number ) {
            printf("You guessed my number in %d guesses.\n", guessCount);
            break;
        } else if (guess > number) {
            printf("Too high. Try again.\n");
        } else {
            printf("Too low. Try again.\n");
        }
    }
}